### PR TITLE
Re-order Build Tasks to Publish to the Ballerina Central 

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
$subject

### CODE REVIEW

File: io-ballerina/build.gradle [hunk 1]
```diff
@@ -245,4 +248,13 @@ ballerinaBuild.dependsOn updateTomlFile
...
 build.dependsOn ballerinaBuild
 
+ballerinaPublish.dependsOn ":${packageName}-native:build"
+ballerinaBuild.dependsOn ":${packageName}-native:test"
+ballerinaBuild.dependsOn ":${packageName}-test-utils:build"
+ballerinaPublish.dependsOn ballerinaBuild
+ballerinaPublish.dependsOn updateTomlFile
+ballerinaPublish.dependsOn initializeVariables
+ballerinaPublish.finalizedBy revertTomlFile
+publish.dependsOn ballerinaPublish
```
Comment (aashikam) [inline io-ballerina/build.gradle:258]:
```suggestion
ballerinaBuild.dependsOn ":${packageName}-native:test"
ballerinaBuild.dependsOn ":${packageName}-test-utils:build"

ballerinaPublish.dependsOn ":${packageName}-native:build"
ballerinaPublish.dependsOn ballerinaBuild
ballerinaPublish.dependsOn updateTomlFile
ballerinaPublish.dependsOn initializeVariables
ballerinaPublish.finalizedBy revertTomlFile
publish.dependsOn ballerinaPublish
```

Comment (BuddhiWathsala) [reply io-ballerina/build.gradle:258]: @aashikam 

```
ballerinaBuild.dependsOn ":${packageName}-native:test"
ballerinaBuild.dependsOn ":${packageName}-test-utils:build"
```
 is alreadly there.

Comment (aashikam) [reply io-ballerina/build.gradle:258]: I just reordered it.

Comment (aashikam) [reply io-ballerina/build.gradle:258]:
```
ballerinaBuild.dependsOn ":${packageName}-native:test"
ballerinaBuild.dependsOn ":${packageName}-test-utils:build"
``` were among `ballerinaPublish` tasks. I took them out. For more readability

File: io-ballerina/build.gradle [hunk 2]
```diff
@@ -245,4 +248,14 @@ ballerinaBuild.dependsOn updateTomlFile
 ballerinaBuild.finalizedBy revertTomlFile
 build.dependsOn ballerinaBuild
 
+ballerinaBuild.dependsOn ":${packageName}-native:test"
+ballerinaBuild.dependsOn ":${packageName}-test-utils:build"
```
[the remark this answers is not in the document]
Comment (aashikam) [reply io-ballerina/build.gradle:252]: This was the only change committed. 7e5ba8358d6c167e0967a9f07951e6877fba83d6

[the remark this answers is not in the document]
Comment (aashikam) [reply io-ballerina/build.gradle:252]: You have added the redundant lines in 2d1cc750a12db33ea257abaf0216020818ca6d62

[the remark this answers is not in the document]
Comment (BuddhiWathsala) [reply io-ballerina/build.gradle]: Yes my mistake, fixed in here https://github.com/ballerina-platform/module-ballerina-io/pull/88/commits/40fcbd441c9adc3607776669b283ff830cd73416

File: io-ballerina/build.gradle [hunk 3]
```diff
@@ -245,4 +248,14 @@ ballerinaBuild.dependsOn updateTomlFile
 ballerinaBuild.finalizedBy revertTomlFile
 build.dependsOn ballerinaBuild
 
+ballerinaPublish.dependsOn ":${packageName}-native:test"
+ballerinaPublish.dependsOn ":${packageName}-test-utils:build"
+
```
Comment (aashikam) [inline io-ballerina/build.gradle:253]:
```suggestion
```

✌️

**Review by aashikam — COMMENTED** [review]
LGTM

**Review by ThisaruGuruge — COMMENTED** [review]
Shall we add the exception when the token is not present?

File: io-ballerina/build.gradle [hunk 4]
```diff
@@ -177,6 +180,9 @@ task ballerinaBuild {
 
 task ballerinaPublish {
     doLast {
+        if (project.version.endsWith('-SNAPSHOT')) {
+            return
+        }
```
Comment (ThisaruGuruge) [inline io-ballerina/build.gradle:185]: Shall we add the exception when the token is not available?

**Review by ThisaruGuruge — APPROVED** [review]

File: io-ballerina/build.gradle (hunk 4 above)
Comment (BuddhiWathsala) [reply io-ballerina/build.gradle:185]: Added in here https://github.com/ballerina-platform/module-ballerina-io/pull/88/commits/6f0645918f8883296f0941f6d4cf3641eb8fa31e

**Pull request merged**